#! /bin/sh -e

# DP: Scratchbox cs2005q3.2-glibc2.5-arm toolchain (used on maemo Diablo), when
# DP: used to compile Python with the default -O3 optimization level, generates
# DP: code in such a way the qemu-arm emulates it incorrectly. This causes
# DP: build failure, so until qemu is fixed, decrease optimization level to
# DP: -O2.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        rm -f configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

---
 configure.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: configure.in
===================================================================
--- configure.in.orig
+++ configure.in
@@ -769,11 +769,11 @@ then
 		# debug builds.
 		OPT="-g -Wall $STRICT_PROTO"
 	    else
-		OPT="-g $WRAP -O3 -Wall $STRICT_PROTO"
+		OPT="-g $WRAP -O2 -Wall $STRICT_PROTO"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall $STRICT_PROTO"
+	    OPT="-O2 -Wall $STRICT_PROTO"
 	    ;;
 	esac
 	case $ac_sys_system in
